File: VideoLocker/src/main/java/org/edx/mobile/module/notification/DummyNotificationDelegate.java
package org.edx.mobile.module.notification;

import org.edx.mobile.model.api.EnrolledCoursesResponse;

import java.util.List;

public class DummyNotificationDelegate implements NotificationDelegate{
    @Override
    public void unsubscribeAll() {

    }

    @Override
    public void resubscribeAll() {

    }

    @Override
    public void syncWithServerForFailure() {

    }

    @Override
    public void checkCourseEnrollment(List<EnrolledCoursesResponse> responses) {

    }

    @Override
    public void changeNotificationSetting(String courseId, String channelId, boolean subscribe) {

    }


    @Override
    public boolean isSubscribedByCourseId(String channel){
        return false;
    }

    @Override
    public void checkAppUpgrade(){}
}

File: OpenEdXMobile/src/main/java/org/edx/mobile/module/notification/DummyNotificationDelegate.kt
package org.edx.mobile.module.notification

import org.edx.mobile.model.api.EnrolledCoursesResponse
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class DummyNotificationDelegate @Inject constructor() : NotificationDelegate {
    override fun unsubscribeAll() {}
    override fun resubscribeAll() {}
    override fun syncWithServerForFailure() {}
    override fun checkCourseEnrollment(responses: List<EnrolledCoursesResponse>) {}
    override fun changeNotificationSetting(
        courseId: String,
        channelId: String,
        subscribe: Boolean
    ) {
    }

    override fun isSubscribedByCourseId(channel: String): Boolean {
        return false
    }

    override fun checkAppUpgrade() {}
}
